js/main.js

```diff
@@ -18,16 +19,42 @@ function submitForm(event) {
     notes: formElements.elements.notes.value,
     entryId: data.nextEntryId,
   };
-  data.nextEntryId++;
-  data.entries.unshift(formEntries);
-  imgElement.src = '../images/placeholder-image-square.jpg';
-  form.reset();
-  const unorderedList = document.querySelector('ul');
-  unorderedList.appendChild(renderEntry(formEntries));
-  viewSwap('entries');
-  if (localStorage.getItem('form-submission') === null) {
-    toggleNoEntries();
+
+  if (data.editing === null) {
+    data.nextEntryId++;
+    data.entries.unshift(formEntries);
+    imgElement.src = '../images/placeholder-image-square.jpg';
+    form.reset();
+    const unorderedList = document.querySelector('ul');
+    unorderedList.appendChild(renderEntry(formEntries));
+    viewSwap('entries');
+    if (localStorage.getItem('form-submission') === null) {
+      toggleNoEntries();
+    }
```

Comment:
is this necessary?

 all the `localStorage` calls should be in `data.js` - refer to the `data` object in the `main.js` file if you need to.